src/chatlog/chatmessage.cpp

```diff
@@ -175,6 +181,72 @@ void ChatMessage::hideDate()
         c->hide();
 }
 
+QString ChatMessage::detectMarkdown(const QString &str)
+{
+    QString out;
+
+    // Create regex for certain markdown syntax
+    QRegExp exp("(\\*\\*)([^\\*\\*]{2,})(\\*\\*)"   // Bold    **text**
+                "|(\\*)([^\\*]{2,})(\\*)"           // Bold *text*
+                "|(\\_)([^\\_]{2,})(\\_)"           // Italics _text_
+                "|(\\_\\_)([^\\_\\_]{2,})(\\_\\_)"  // Italics __text__
```

Comment:
Well, problem is that it's supposed to be a markdown support. As long as qTox and other clients actually do the markdown, there'll be no problem with how formatted text looks. If qTox will start to parsing text in arbitrary way.. eh.

So, either you do the real markdown, and call it basic markdown support, or you're venturing into area where there are no standards, and there is lots of confusion and non-standard behaviour.

Also, note that while people who used skype might be used to that (I didn't even know that skype supported text parsing, spare for emoticons), note that large parts of the interned actually do support markdown - e.g. reddit.